rust/connlib/tunnel/src/client.rs

```diff
@@ -1152,12 +1161,64 @@ mod tests {
         assert_eq!(client_state.poll_event(), Some(Event::RefreshInterface));
     }
 
+    #[test]
+    fn sentinel_dns_works() {
+        let servers = dns_list();
```

Comment:
If a test relies on some state being set up, it would be nice to make that obvious and pass it in from the top level. That makes it possible to glance at the test and check what it does :)

Also, splitting it in roughly 3 parts: arrange, act, assert would be nice :)

Reply:
> pass it in from the top level

Like the 3 strings for the IPs.